### ISSUE TITLE
byte[] is not deserialized by SoapSerializer.XmlSerializer

### ISSUE THREAD

**Issue by IvanLeonenko** [DigDes/SoapCore#625]
I'm using `SoapSerializer.XmlSerializer` and trying to get simple web method to work (see below).

Basically, I want to pass byte array. I see that in generated wsdl type for the parameter is correct - `s:base64Binary`.

But when I try to send request (inline base64 payload dGVzdCBzdHJpbmc=) from generated client code or from SoapUI I get the following error:
```
    <s:Fault>
      <faultcode>s:Client</faultcode>
      <faultstring>End element 'data' from namespace 'http://test/testsvc' expected. Found text 'dGVzdCBzdHJpbmc='. Line 1, position 231.</faultstring>
    </s:Fault>
```

Same request works fine in old .NET service, no erros.

Request
```
<soapenv:Envelope xmlns:soapenv="http://schemas.xmlsoap.org/soap/envelope/" xmlns:tes="http://test/testsvc">
   <soapenv:Header/>
   <soapenv:Body>
      <tes:Test>
         <tes:data>dGVzdCBzdHJpbmc=</tes:data>
      </tes:Test>
   </soapenv:Body>
</soapenv:Envelope>
```

Service
```
[ServiceContract(Namespace = "http://test/testsvc")]
    public interface ITestService
    {
        [OperationContract]
        ResultResponse Test(byte[] data);
    }

    public class ResultResponse
    {
        public bool Result { get; set; }
        public string Base64 { get; set; }
    }

    public class TestService : ITestService
    {
        public ResultResponse Test(byte[] data)
        {
            return new ResultResponse() {Result = true, Base64 = Convert.ToBase64String(data)};
        }
    }
```

I noticed that in `SerializerHelper` there's [a check](https://github.com/DigDes/SoapCore/blob/491bd57c138eb6074ad54433a8fb652dc92c53b9/src/SoapCore/SerializerHelper.cs#L38)  after which Byte[] is handled in `DeserializeArrayXmlSerializer`  and generates the error. If I call `DeserializeObject` for Byte[] type instead of `DeserializeArrayXmlSerializer` then parameter is deserialized correctly.

Is it a bug, or I'm doing something wrong?

Thank you

**Comment by IvanLeonenko** [comment]
#626

**Comment by kotovaleksandr** [comment]
fixed in #626

**Comment by samuel-barbosa** [comment]
I could see that the fix was really effective for methods with _single dimension_ array of byte arguments.

However, in my project there are some methods with _multi dimension_ arrays in theirs arguments. 
When I try to call them, I get the same message error reported above (_End element '<my argument name>' from namespace 'http://mynamespace' expected. Found element 'base64Binary' from namespace 'http://mynamespace'. Line 1, position 249_).

These are my Web Methods:
![Screen001](https://user-images.githubusercontent.com/35376638/121594694-cc8e4f80-ca13-11eb-9f7a-037bc4bd761b.png)

When calling the first one using Postman, it works:
![Screen002](https://user-images.githubusercontent.com/35376638/121594853-02333880-ca14-11eb-8849-2b78f8d2cd78.png)

When calling that with two dimensions, it fails:
![Screen003](https://user-images.githubusercontent.com/35376638/121595116-550cf000-ca14-11eb-90c8-cebcd04dacf9.png)

I´m using version 1.1.0.9

Thanks
